test/functional/test_framework/ellsq.py

```diff
@@ -0,0 +1,243 @@
+    [(fe(0xa91b7f2ab93de821abeec1750258e4d4f5f09831b0a11dda47e89ddf6944d819), fe(0x22eb9bf64a517df2c27d1c551df07609166fc995e2b39fee0473ea46ed14efc1)), [fe(0x7a01651a81a7f09e2733cf349e6472a118c167806f5c880f534b89a652be06a7), None                                                                  , None                                                                  , None                                                                  ]],
+    [(fe(0x0000000000000000000000000000000000000000000000000000000000000001), fe(0x4218f20ae6c646b363db68605822fb14264ca8d2587fdd6fbc750d587e76a7ee)), [fe(0x2c8864a8c34e87d753ee73008bbed54a47b3790756d0b74710341b37f598a5fe), fe(0x15908d622377bedc0fecf55fcc6425c9de992fcb01af2628ac40f22088de01f0), None                                                                  , None                                                                  ]],
+    [(fe(0xa64de96a6254cefcffbeaf898f2c228af6d405f3bcc6a4cce068312af7ccf8e1), fe(0x8f9b3a1b2d146ea954bfc5e2cdfe861ccbed8431c741c5f9d32f16a3073ea496)), [fe(0x4591d33d1a133a8794689b1b0ca445b78ada3bcec2e812b08315e2b107940ad4), fe(0xa763d2176027d40e8a8ff34bd9c639b73e2ea04592274fdcfa4051c66d93a1b6), None                                                                  , None                                                                  ]],
+    [(fe(0x49a0dc068c3f117aefdc842d3d358153f677f04c6dabc9c91b09d452fef27b66), fe(0x7b944da48a175dbc444ead8db82eff66b081a8aae6453fed2bca9720b44dd6e5)), [fe(0x7bf1e2b1720c1c440db64687f16439fa41b398338095f24ebeec0cfa88750dc9), fe(0xdc97e26d3137445d6c1269b61a7655010c19c36a2e361066e31e2bb10403470b), None                                                                  , None                                                                  ]],
+    [(fe(0xd09a4047f158fe52f96c661d02c68657c4c976ea96ea85ef46d6985bd540756b), fe(0xe793bfaae9300f18e6f9b55aae26322368b61d51ae5022efe266c72d574178bc)), [fe(0x7e6175fdfbb9fb4faf6e2b925ef86c4a444d819aaa82dbee545d3d9b296375be), None                                                                  , None                                                                  , None                                                                  ]],
+    [(fe(0x3498662504b73c7c8cecb6c33cd493bdfc190e0f87d913d7ff9ad42e222bfe95), fe(0x245b3a61b8d46997f14f2fea2874899691eb32542b9907d65eb9d21d42454021)), [fe(0x7f556282c3dd9d263390d6bbddada698ab8fd7c7d1a06498f42b30437c8361ad), None                                                                  , None                                                                  , None                                                                  ]]
+]
+
+def encode(P):
+    while True:
+        u = fe(random.randrange(1, SECP256K1_ORDER))
+        fe1 = f(u)
```

Comment:
Using `fe1` as variable name here is a bit confusing, as the return value is a *pair* of fe values (the x and y coordinates of a point on the curve). Use `p` or `ge` or so?